atlasdb-config/src/main/java/com/palantir/atlasdb/factory/AtlasDbServiceDiscovery.java

```diff
@@ -37,6 +38,15 @@ public static DbTimeLockFactory createDbTimeLockFactoryOfCorrectType(KeyValueSer
         return createAtlasDbServiceOfCorrectType(config, DbTimeLockFactory::getType, DbTimeLockFactory.class);
     }
 
+    public static NamespaceDeleterFactory createNamespaceDeleterFactoryOfCorrectType(KeyValueServiceConfig config) {
+        if (!config.enableNamespaceDeletion()) {
+            throw new SafeIllegalStateException("Cannot construct a NamespaceDeleterFactory when keyValueService"
+                    + ".enableNamespaceDeletionDangerousIKnowWhatIAmDoing is false.");
+        }
```

Comment:
I'm a touch uncertain about throwing: is it only the case that we'll provide this lazily? If yes, fine, but if no, won't this just make AtlasDB throw for those using dagger and trying to get the factory?